samples/tiff-test.cpp

```diff
@@ -37,29 +37,24 @@ void mini1(const char* path)
 {
     ExifData exifData;
     Blob blob;
-    WriteMethod wm;
 
     // Write nothing to a new structure, without a previous binary image
-    wm = ExifParser::encode(blob, 0, 0, bigEndian, exifData);
-    assert(wm == wmIntrusive);
```

Comment:
If I understand the code correctly, then `ExifParser::encode` can actually return different values then `wmIntrusive`. I guess this was intended as a meaningfull test, that the write method was e.g. not `wmNonIntrusive`.

(Of course this whole file should rather be converted into a unit test instead of using manual asserts.)

Reply:
This is what I thought when I decided to remove the asserts and the return type. To me, those assertions where there just to show an example and document what the code is returning. When we continue writing unit tests for the exiv2lib, these tests applications will become irrelevant.

Reply:
Yes, but my point is that the return value is probably meaningful, even if this code is only a demonstration/example. If we remove the assert, we'll forget about the return value when we port this file to gtest.

Reply:
Would it be ok for you to replace those `asserts` by `enforce` ? Or would you prefer to use the `UNUSED()` trick ?

Reply:
Ah, sorry, I kind of didn't get the point of removing the `assert`. But I just remembered that cmake defines `NDEBUG` in Release builds and then the variable is indeed useless.

`enforce` is probably the way to go here for now.

(I just realized that we then did not test this stuff in Release builds...)

Reply:
We are using Release builds in Travis and Appveyor. That's indeed one of the reasons why I want to clean the warnings. In that way we will have builds free of warnings.

Alright, I will use `enforce`here ;)